src/InvestJob.jsx

```diff
@@ -73,59 +90,68 @@ export class InvestJob extends React.Component {
     this.setSessionID = this.setSessionID.bind(this);
   }
 
-  saveState(event) {
-    // Save a snapshot of this component's state to a JSON file.
+  saveState() {
+    /** Save the state of the application (1) and the current InVEST job (2).
+    * 1. Save the state object of this component to a JSON file .
+    * 2. Append metadata of the invest job to a persistent database/file.
+    * This triggers automatically when the invest subprocess starts and again
+    * when it exits.
+    */
+    const jobName = this.state.sessionID;
     const jsonContent = JSON.stringify(this.state, null, 2);
-    const sessionID = this.state.sessionID;
-    const filepath = path.join(CACHE_DIR, sessionID + '.json');
+    const filepath = path.join(CACHE_DIR, jobName + '.json');
     fs.writeFile(filepath, jsonContent, 'utf8', function (err) {
       if (err) {
         console.log("An error occured while writing JSON Object to File.");
         return console.log(err);
       }
-      console.log("saved" + sessionID);
+      console.log("saved: " + jobName);
     });
-    this.props.updateRecentSessions(sessionID);
+    let job = {};
+    job[jobName] = {
+      model: this.state.modelName,
+      workspace: this.state.workspace,
+      statefile: filepath,
+      status: this.state.jobStatus,
+      humanTime: new Date().toLocaleString(),
+      systemTime: new Date().getTime(),
+    }
+    this.props.updateRecentSessions(job, this.props.appdata);

```

Comment:
That's right. `updateRecentSessions` will overwrite an existing entry with an identical `sessionID`, and that's intended since invest will also overwrite that model run. I think this okay for now, but if someday invest caches some interesting data that is not over-written when a workspace + suffix are re-used, then we could revisit this.

I'm pretty sure we're going to revisit the structure of this job metadata and the fact that it's just stored in a json file anyway. This is one of the most experimental parts of the UI at this point.

Reply:
Also in your example above: "what would happen if we re-run the same model in the same directory with the same suffix, but with different inputs?"

If a user did overwrite a previous run, but then wanted to redo that original run, they would no longer be able to get there by loading a "recent session", but they would still have the invest logfile from the earlier run and could load that via "Load Parameters" button.